scripts/tests/test_zkai_native_seq32_attention_mlp_adapter_variant_selector_gate.py

```
@@ -0,0 +1,133 @@
+    def test_mutation_inventory_is_exact(self) -> None:
+        result = self.__class__.payload["mutation_result"]
+        self.assertTrue(result["all_mutations_rejected"])
+        self.assertEqual(result["mutations_rejected"], len(self.gate.MUTATION_NAMES))
+        self.assertEqual(tuple(result["mutation_names"]), self.gate.MUTATION_NAMES)
+        self.assertEqual(tuple(case["name"] for case in result["cases"]), self.gate.MUTATION_NAMES)
+
+    def test_rejects_best_variant_gap_erasure(self) -> None:
+        payload = copy.deepcopy(self.__class__.payload)
+        payload["summary"]["best_variant_gap_typed_bytes"] = 0
+        payload["payload_commitment"] = self.gate.payload_commitment(payload)
+        with self.assertRaisesRegex(self.gate.NativeSeq32AdapterVariantSelectorGateError, "summary drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)